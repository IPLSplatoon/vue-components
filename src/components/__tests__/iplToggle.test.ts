import IplToggle from '../iplToggle.vue';
import { shallowMount } from '@vue/test-utils';

describe('IplToggle', () => {
    it('matches snapshot', () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: false } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot with custom labels', () => {
        const wrapper = shallowMount(IplToggle, {
            props: {
                modelValue: false,
                trueLabel: 'Enable',
                falseLabel: 'Disable'
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('toggles value on click', async () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: false } });

        await wrapper.find('.ipl-toggle__container').trigger('click');

        const emitted = wrapper.emitted()['update:modelValue'];
        expect(emitted.length).toEqual(1);
        expect(emitted[0]).toEqual([ true ]);
    });

    it('gives class to true option text when value is true', () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: true } });

        expect(wrapper.find('.true-option').classes()).toContain('selected');
        expect(wrapper.find('.false-option').classes()).not.toContain('selected');
    });

    it('gives class to false option text when value is false', () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: false } });

        expect(wrapper.find('.true-option').classes()).not.toContain('selected');
        expect(wrapper.find('.false-option').classes()).toContain('selected');
    });

    it('gives class to button element if value is true', () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: true } });

        expect(wrapper.find('.ipl-toggle__button').classes()).toContain('is-true');
    });

    it('does not give class to button element if value is false', () => {
        const wrapper = shallowMount(IplToggle, { props: { modelValue: false } });

        expect(wrapper.find('.ipl-toggle__button').classes()).not.toContain('is-true');
    });
});
