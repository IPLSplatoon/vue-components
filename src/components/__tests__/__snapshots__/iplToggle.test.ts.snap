// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplToggle matches snapshot 1`] = `
"<div class=\\"ipl-toggle__container\\">
  <div class=\\"ipl-toggle__text\\"><span class=\\"true-option\\">Show</span><span class=\\"false-option selected\\">Hide</span></div>
  <div class=\\"ipl-toggle__button\\"></div>
</div>"
`;

exports[`IplToggle matches snapshot with custom labels 1`] = `
"<div class=\\"ipl-toggle__container\\">
  <div class=\\"ipl-toggle__text\\"><span class=\\"true-option\\">Enable</span><span class=\\"false-option selected\\">Disable</span></div>
  <div class=\\"ipl-toggle__button\\"></div>
</div>"
`;
